### PR TITLE
Refactor to remove `_.cloneDeep`

### PR DESCRIPTION
**Pull request by stephenwade**
> Which issue, if any, is this issue related to?

Ref #4412.

> Is there anything in the PR that needs further explanation?

No.

### CODE REVIEW

**Comment by ybiquitous** [comment]
@stephenwade Thank you for creating this PR! The `deep-clone` package seems good to me. 👍🏼 

However, I doubt that `_.cloneDeep` is necessary in the first place. 🤔 

When I read the `needlessDisables()` function carefully, `_.cloneDeep` seems necessary to change the `rangeData` object:

https://github.com/stylelint/stylelint/blob/dadcf7b407ed5cb09cac70e943713f94b4fae3a0/lib/needlessDisables.js#L58-L68

I think we may be able to replace the logic above as below:

```diff
+		const allRangeComments = new Set(rangeData.all.map(range => range.comment));
+
 		for (const [rule, ranges] of Object.entries(rangeData)) {
 			for (const range of ranges) {
+				if (rule !== 'all' && allRangeComments.has(range.comment)) continue;
+
 				if (enabled === optionsMatches(options, 'except', rule)) continue;
 
 				const useful = usefulDisables.get(range.comment) || new Set();
```

What do you think?

If we could stop using `_.cloneDeep`, we could avoid the useless copy of big objects (`stylelint.DisabledRange`) including a `postcss.Comment` node.

**Comment by stephenwade** [comment]
@ybiquitous I have modified the code like you suggested. I also found a way to use `deepmerge` to replace the other use of `_.cloneDeep` so I can avoid adding the dependency at all.

File: lib/needlessDisables.js [hunk 1]
```diff
@@ -22,7 +21,7 @@ module.exports = function (results) {
 		const [enabled, options, stylelintResult] = settings;
 
 		/** @type {{[ruleName: string]: Array<DisabledRange>}} */
```
Comment (ybiquitous) [inline lib/needlessDisables.js:23]: [trivial] This `@type` annotation seems needless thanks to the type inference:

```suggestion
```

**Review by ybiquitous — APPROVED** [review]
@stephenwade Thank you for addressing my suggestion! I‘ve left a trivial comment, but this change looks good to me. 👍🏼

File: lib/needlessDisables.js (hunk 1 above)
Comment (stephenwade) [reply lib/needlessDisables.js]: @ybiquitous You're right, I think that type must have been needed when we were deep copying, but it is not important anymore.

**Review by ybiquitous — APPROVED** [review]
@stephenwade Thanks! LGTM 👍🏼

**Pull request merged**